nimare/nimads.py

```diff
@@ -0,0 +1,301 @@
+"""NIMADS-related classes for NiMARE."""
+import json
+
+
+class Studyset:
+    """A collection of studies for meta-analysis.
+
+    This is the primary target for Estimators and Transformers in NiMARE.
+
+    Attributes
+    ----------
+    studies : list of Study objects
+        The Study objects comprising the Studyset.
+    """
+
+    def __init__(self, source, target_space=None, mask=None):
+        self.id = source["id"]
+        self.name = source["name"] or ""
+        self.studies = [Study(s) for s in source["studies"]]
+        self._annotations = []
+
+    def __repr__(self):
+        """My Simple representation."""
+        return repr("Studyset: " + self.id)
+
+    def __str__(self):
+        """Give useful information about the Studyset."""
+        return str(" ".join(["Studyset:", self.name, "::", f"studies: {len(self.studies)}"]))
+
+    @property
+    def annotations(self):
+        """Return existing Annotations."""
+        return self._annotations
+
+    @annotations.setter
+    def annotations(self, annotation):
+        # some logic to compare ids
+        # ss_analysis_ids = set([a.id for s in self.studies for a in s.analyses])
+        # annot_analysis_ids = set([n['analysis'] for n in annotation.notes])
+        self._annotations.append(annotation)
+
+    @annotations.deleter
+    def annotations(self, annotation_id=None):
+        if annotation_id:
+            self._annotations = [a for a in self._annotations if a.id != annotation_id]
+        else:
+            self._annotations = []
+
+    @classmethod
+    def from_nimads(cls, filename):
+        """Create a Studyset from a NIMADS JSON file."""
+        with open(filename, "r+") as fn:
+            nimads = json.load(fn)
+
+        return cls(nimads)
+
+    def to_nimads(self, filename):
+        """Write the Studyset to a NIMADS JSON file."""
+        ...
+
+    def load(self, filename):
+        """Load a Studyset from a pickled file."""
+        ...
+
+    def save(self, filename):
+        """Write the Studyset to a pickled file."""
+        ...
+
+    def copy(self):
+        """Create a copy of the Studyset."""
+        ...
+
+    def slice(self, analyses):
+        """Create a new Studyset with only requested Analyses."""
+        ...
+
+    def merge(self, right):
+        """Merge a separate Studyset into the current one."""
+        ...
+
+    def update_image_path(self, new_path):
+        """Point to a new location for image files on the local filesystem."""
+        ...
+
+    def get_analyses_by_coordinates(self, xyz, r=None, n=None):
```

Comment:
The `get*` methods can be removed, since they'll be in `DatasetSearcher` in the future.